tests/test_vfs.py

```diff
@@ -83,13 +83,3 @@ def test_read_vfs_none():
             'tests/data/RGB.byte.tif') as src:
         assert src.name == 'tests/data/RGB.byte.tif'
         assert src.count == 3
-
-
-@pytest.mark.parametrize('mode', ['r+', 'w'])
-def test_update_vfs(tmpdir, mode):
-    """VFS datasets can not be created or updated"""
-    with pytest.raises(TypeError):
```

Comment:
They do. I've glossed over the implications of this. Does it expose users to a bunch of VSI-specific details (zip vs https, etc) and poke holes in our abstraction?